### PR TITLE
STM32MP1 LmP OP-TEE/Linux

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 1]
```diff
@@ -430,7 +432,7 @@ EXTRA_IMAGEDEPENDS:remove:stm32mp15-disco = "sdcard-raw-tools-native"
 WKS_FILE_DEPENDS:append:stm32mp15-disco = " u-boot-default-script"
 PREFERRED_PROVIDER_u-boot-default-script:stm32mp15-disco = "u-boot-ostree-scr-fit"
 WKS_FILE_DEPENDS:remove:stm32mp15-disco = "st-image-bootfs st-image-userfs"
-WKS_FILE:sota:stm32mp15-disco = "sdimage-stm32mp157c-dk2-trusted-sota.wks.in"
+WKS_FILE:sota:stm32mp15-disco = "sdimage-stm32mp157c-dk2-optee-sota.wks.in"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:439]: This wks file will need to be renamed as well.

File: meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend [hunk 2]
```diff
@@ -1 +1,20 @@
 require optee-os-fio-bsp.inc
+
+OPTEE_SUFFIX:stm32mp1common    = "${@bb.utils.contains('MACHINE_FEATURES','fip','bin','stm32',d)}"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:3]: I thinkwe only support fip, right?

File: meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend [hunk 3]
```diff
@@ -1 +1,20 @@
...
+OPTEE_HEADER:stm32mp1common    = "tee-header_v2"
+OPTEE_PAGEABLE:stm32mp1common  = "tee-pageable_v2"
+OPTEE_PAGER:stm32mp1common     = "tee-pager_v2"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:6]: In theory these files are already deployed by default at the deploy / machine / optee folder.

Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:6]: fip bbclacs  from `meta-st-stm32mp1` expects this naming format of optee images, so this is why I added that additional `do_deploy()` step:

in `meta-st-stm32mp1/classes/fip-utils-stm32mp.bbclass`:
```
206     elif [ "${bl32_conf}" = "optee" ]; then                             
207                 # Check for files                                               
208                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
209                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
210                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
211                 # Set FIP_EXTRACONF                                             
212                 FIP_EXTRACONF="\                                                
213                     --tos-fw ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX} \
214                     --tos-fw-extra1 ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX} \
215                     --tos-fw-extra2 ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX} \
216                     "                                                           

```

File: meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend (hunk 2 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:3]: yes, I can just drop that

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
Comment (igoropaniuk) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:439]: I just forgot to push new wks.in file.

**Review by angolini — APPROVED** [review]

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 4]
```diff
@@ -393,7 +393,7 @@ WKS_FILE:sota:imx8mp-lpddr4-evk-ebbr ?= "efidisk-sota.wks"
 MACHINE_FEATURES:append:imx8mp-lpddr4-evk = " nxp89xx mxm-mwifiex-load"
 
 # STM32MP1
-PREFERRED_PROVIDER_virtual/kernel:stm32mp1common ?= "linux-lmp-stm32"
+PREFERRED_PROVIDER_virtual/kernel:stm32mp1common ?= "linux-lmp"
```
Comment (ricardosalveti) [inline meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:396]: We should include the meta bump commit as part of this pr as well.

Comment (igoropaniuk) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:396]: done

File: meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend (hunk 3 above)
Comment (ricardosalveti) [reply meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:6]: Sure, but then we just need the if block, not else, as those files are already created and deployed by default (with similar name).

File: meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend [hunk 5]
```diff
@@ -1 +1,20 @@
...
+    if [ -n "${OPTEE_CONF}" ]; then
+        for conf in ${OPTEE_CONF}; do
+            install -m 644 ${B}/core/${OPTEE_HEADER}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_HEADER}-${conf}.${OPTEE_SUFFIX}
+            install -m 644 ${B}/core/${OPTEE_PAGER}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_PAGER}-${conf}.${OPTEE_SUFFIX}
+            install -m 644 ${B}/core/${OPTEE_PAGEABLE}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_PAGEABLE}-${conf}.${OPTEE_SUFFIX}
+        done
+    else
+        install -m 644 ${B}/core/${OPTEE_HEADER}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_HEADER}.${OPTEE_SUFFIX}
+        install -m 644 ${B}/core/${OPTEE_PAGER}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_PAGER}.${OPTEE_SUFFIX}
+        install -m 644 ${B}/core/${OPTEE_PAGEABLE}.${OPTEE_SUFFIX} ${DEPLOYDIR}/optee/${OPTEE_PAGEABLE}.${OPTEE_SUFFIX}
```
Comment (ricardosalveti) [inline meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend:18]: e.g. from kria (what I have locally):
```
rsalveti@evatuf:~/build/lmp/build-lmp$ ls -l deploy/images/kv260/optee/
total 4684
-rw-r--r-- 1 rsalveti rsalveti  416100 Jun  8 13:42 tee.bin
-rw-r--r-- 1 rsalveti rsalveti 4012464 Jun  8 13:42 tee.elf
-rw-r--r-- 1 rsalveti rsalveti      28 Jun  8 13:42 tee-header_v2.bin
-rw-r--r-- 1 rsalveti rsalveti      11 Jun  8 13:42 tee-init_load_addr.txt
-rw-r--r-- 1 rsalveti rsalveti       0 Jun  8 13:42 tee-pageable_v2.bin
-rw-r--r-- 1 rsalveti rsalveti  416072 Jun  8 13:42 tee-pager_v2.bin
```

So I think we are safe to just keep the if block.

**Review by ricardosalveti — APPROVED** [review]
LGTM

**Pull request merged**